yuptoo/modifiers/transform_mac_addresses.py

```diff
@@ -1,7 +1,26 @@
-from yuptoo.processor.utils import Modifier
+import copy
 import logging
+from yuptoo.processor.utils import Modifier
 
 LOG = logging.getLogger(__name__)
+TRANSFORMED_DICT = dict({'removed': [], 'modified': [], 'missing_data': []})
+
+
+def _remove_mac_addrs_for_omitted_nics(
+    host: dict, mac_addresses_to_omit: list,
+        transformed_obj=copy.deepcopy(TRANSFORMED_DICT)):
+    """Remove mac_addresses for omitted nics."""
+
+    mac_addresses = host.get('mac_addresses')
+    len_of_mac_addrs_to_omit = len(mac_addresses_to_omit)
+    if mac_addresses and len_of_mac_addrs_to_omit > 0:
+        host['mac_addresses'] = list(
+            set(mac_addresses) - set(mac_addresses_to_omit))
+        if not host['mac_addresses']:
```

Comment:
`and` operator (line 16) already covers `!host[''mac_addresses]`. So, the statement under this `if` (line 19) won't be executed anyway. Is my assumption correct? Can we align it with the `if` statement (at line 16)?

Reply:
Intention behind this line is when there is empty list as result of difference between host's current mac_addresses values & mac_addresses_to_omit.

Reply:
Got it 👍🏻 Thanks!